### PR TITLE
Refactor/justinyaodu/1058459663 add batchid

### PR DESCRIPTION
**Pull request by justinyaodu**
### Administrative Info
Monday Board ID: 1058459663
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[Monday Item ID]-[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes
What changes did you make?
- Added the `batchId` field to the Mongoose schema, and modified the indexing to enforce that the batch IDs are unique.

### Testing
How did you confirm your changes worked? 
- We think the code looks good, but we can't test it without changing the routes, which is going to be done separately. If there are issues we can fix them then.

### Confirmation of Change 
Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.
- N/A

### CODE REVIEW

File: client/src/MongodbFunctions.js [hunk 1]
```diff
@@ -8,14 +8,13 @@ const BACKEND_URL = 'http://localhost:9000';
...
  */
-export const postNewItem = async (token, itemName, itemId, weight, outDate) => {
+export const postNewItem = async (token, itemName, weight, outDate) => {
   try {
     const response = await fetch(BACKEND_URL + '/items/addItem', {
       method: 'POST',
       headers: {'Content-Type': 'application/json', token: token},
       body: JSON.stringify({
         itemName: itemName,
-        itemId: itemId,
```
Comment (hahsu) [inline client/src/MongodbFunctions.js:18]: won't we need batchId here?

File: client/src/MongodbFunctions.js [hunk 2]
```diff
@@ -8,14 +8,13 @@ const BACKEND_URL = 'http://localhost:9000';
  * @param {string} outDate The date that this batch should be sent by.
```
Comment (hahsu) [inline client/src/MongodbFunctions.js:8]: can we update the comments, line 6 still has itemID

File: backend/routes/items.js [hunk 3]
```diff
@@ -81,7 +80,7 @@ router.post("/addItem", token_required, async (req, res, next) => {
     return;
   }
 
-  const { itemName, itemId, weight } = req.body;
```
Comment (hahsu) [inline backend/routes/items.js:84]: won't we need batchId here?

**Review by hahsu — CHANGES_REQUESTED** [review]

**Review by hahsu — APPROVED** [review]
lgtm, thank you!

**Review by petabite — APPROVED** [review]
Looks good!

**Pull request merged**